routers/web/repo/view.go

```diff
@@ -151,8 +151,8 @@ func localizedExtensions(ext, languageCode string) (localizedExts []string) {
 	if strings.Contains(lowerLangCode, "-") {
 		underscoreLangCode := strings.ReplaceAll(lowerLangCode, "-", "_")
 		indexOfDash := strings.Index(lowerLangCode, "-")
-		// e.g. [.zh-cn.md, .zh_cn.md, .zh.md, .md]
-		return []string{lowerLangCode + ext, underscoreLangCode + ext, lowerLangCode[:indexOfDash] + ext, ext}
+		// e.g. [.zh-cn.md, .zh_cn.md, .zh.md, _zh.md, .md]
+		return []string{lowerLangCode + ext, underscoreLangCode + ext, lowerLangCode[:indexOfDash] + ext, "_" + lowerLangCode[1:indexOfDash] + ext, ext}
```

Comment:
This functions was designed for work for both `languageCode=en-US` and `languageCode=en`.

This change only supports `languageCode=en-US`, while line 158-158 are for `languageCode=en`, the behaviors are inconsistent now.

However, I have no objection to merge this PR since there is no `languageCode=en` in Gitea code, while it's better to document the behavior.

Reply:
Hmm, In my view, the en is same with en-US (or other). in other words, it's not a good idea to add a new file named as `locale_en.ini` in [``i18n folder``](https://github.com/go-gitea/gitea/tree/main/options/locale). and the link 'https://gitea.com/?lang=en' should have same behave with 'https://gitea.com/?lang=en-US'

Reply:
Yup, Gitea doesn't use the `en` or `zh` at the the moment, the code is just there. Line 158-159 won't run in current Gitea.

Reply:
> Hmm, In my view, the `en` is same with en-US (or other). 

There is a complex and special relationship between en, en-US, en_001 and other en subvariants e.g. en_GB

Whilst en-US is the default locale for en, it is not en. For example the rules for dates in en-US are extremely different from those for almost every other en variant. CLDR has en_001 which almost completely overrides most US weirdness and means that most English regional locales inherit from en_001 rather than en directly.

https://cldr.unicode.org/development/development-process/design-proposals/english-inheritance

Our translation scheme isn't really designed at present to properly consider regional localisation but we're probably going to have to consider it at some point soon.